File: Tests/Untitled-2.cpp
#include <iostream>

class Person {
public:
    Person(const std::string &name, int age) : name(name), age(age) {}

    void introduce() const {
        std::cout << "Hello, my name is " << name << " and I am " << age << " years old." << std::endl;
    }

private:
    std::string name;
    int age;
};

int main() {
    Person person("Alice", 30);
    person.introduce();
    return 0;
}

File: Tests/Untitled-2.rs
struct Person {
    name: String,
    age: u32,
}

impl Person {
    fn new(name: &str, age: u32) -> Person {
        Person {
            name: name.to_string(),
            age,
        }
    }

    fn introduce(&self) {
        println!("Hello, my name is {} and I am {} years old.", self.name, self.age);
    }
}

fn main() {
    let person = Person::new("Alice", 30);
    person.introduce();
}
